### PR TITLE
Feat: Wallet Instantiation

### LINKED ISSUE #1: Accounts / Wallet
**Goal 1:** As a developer, I must be able to create a new account in Hedera

**Goal 2:** As a developer, I must be able to access my already existing account in Hedera and use it to sign & submit transactions to Hedera HSCS

**Prerequisite:** [Provider](https://www.notion.so/EthersJs-7f2bc86082ce425bb21e3033b5f39280)

**Description:** `hethers.js` must change the existing `ethers.Signers` interface in order to support Hedera `Account Creation`. The most prominent change is the fact that creating accounts will involve Hedera TX. The rest of the Signer/Wallet interfaces will not be changed drastically. Other notable changes/unsupported APIs:

- `signer.getGasPrice` → Will not be supported at least initially
- `signer.getTransactionCount` → Will not be supported at least initially
- `signer.resolveName` → Will not be supported at least initially
- `signer.signMessage` → Should be discussed
- `signer._signTypedData` → Should be discussed
- `new ethers.Wallet` → Will have different arguments
- `Wallet.CreateRandom` → Will have different arguments
- `Wallet.fromEncryptedJson` & `Wallet.fromEncryptedJsonSync` → Will have different arguments
- `Wallet.fromMnemonic` → Should be discussed
- `wallet.publicKey` → Should be discussed

### LINKED ISSUE #15: Implement signer.getAccount

### LINKED ISSUE #17: Implement creating Signer instance

### LINKED ISSUE #18: Implement creating Wallet instance

### LINKED ISSUE #32: Resolve Wallet failing tests
The goal of this task is to resolve all failing tests for the new implementation of the `wallet`

### PR DESCRIPTION
**Pull request by Daniel-K-Ivanov**
**Description**:
- Adds wallet instantiation by EOA or Hedera Account pair
- Adds new examples for the instantiation
- Updates the util for parsing accounts
- Implements HIP-32

Fixes #17 
Fixes #18
Fixes #1 
Fixes #15 
Fixes #17
Fixes #18
Fixes #32 

**Notes for reviewer**:

**Checklist**
- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/wallet/lib/index.d.ts [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
     readonly address: string;
+    readonly account: Account;
     readonly provider: Provider;
     readonly _signingKey: () => SigningKey;
     readonly _mnemonic: () => Mnemonic;
-    constructor(privateKey: BytesLike | ExternallyOwnedAccount | SigningKey, provider?: Provider);
+    constructor(acc: ExternallyOwnedAccount | HederaAccount, provider?: Provider);
     get mnemonic(): Mnemonic;
     get privateKey(): string;
     get publicKey(): string;
```
Comment (Daniel-K-Ivanov) [inline packages/wallet/lib/index.d.ts:18]: todo add getter for account similar to mnemonic, pk and public key

**Review by joan41868 — APPROVED** [review]
Awesome :)

File: tsconfig.project.json [hunk 2]
```diff
@@ -105,21 +105,21 @@
...
-    },
-    {
-      "path": "./packages/experimental"
-    },
+//    {
+//      "path": "./packages/cli"
+//    },
+//    {
+//      "path": "./packages/experimental"
+//    },
```
[the remark this answers is not in the document]
Comment (joan41868) [reply tsconfig.project.json:113]: As I saw the CI on this branch, I suggest you remove the lines, as this is what fails the build.

**Review by alshapi — APPROVED** [review]

File: packages/tests/src.ts/test-account.spec.ts [hunk 3]
```diff
@@ -5,42 +5,57 @@ import assert from 'assert';
...
-describe('Private key generation', function() {
+describe('Private key generation & Alias population', function() {
     let tests: Array<TestCase> = loadTests('accounts');
     tests.forEach((test) => {
         if (!test.privateKey) { return; }
+
         it(('correctly converts private key - ' + test.name), function() {
             let wallet = new ethers.Wallet(test.privateKey);
-            assert.equal(wallet.address.toLowerCase(), test.address.toLowerCase(),
+            assert.strictEqual(wallet.alias.toLowerCase(), test.alias.toLowerCase(),
```
Comment (joan41868) [inline packages/tests/src.ts/test-account.spec.ts:25]: why do we need to do `toLowerCase` on both? Isn't it meaningless in this case?

**Review by joan41868 — APPROVED** [review]
LGTM

File: packages/tests/src.ts/test-account.spec.ts [hunk 4]
```diff
@@ -5,42 +5,57 @@ import assert from 'assert';
...
 });
 
+describe('Account & Address population', () => {
+    let tests: Array<TestCase> = loadTests('accounts');
+    tests.forEach(test => {
+        if (!(test.address && test.account && test.privateKey)) { return; }
+
+        it(('correctly populates account/address - ' + test.name), () => {
+            let wallet = new ethers.Wallet({privateKey: test.privateKey, address: test.address});
+            assert.strictEqual(wallet.address.toLowerCase(), test.address.toLowerCase(), `correctly populates address - ` + test.address);
```
Comment (alshapi) [inline packages/tests/src.ts/test-account.spec.ts:38]: same, `toLowerCase` on both?

**Review by alshapi — APPROVED** [review]

**Pull request merged**